src/main/kotlin/org/rust/ide/statistics/RsCodeVisionUsageCollector.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.statistics
+
+import com.intellij.internal.statistic.eventLog.EventLogGroup
+import com.intellij.internal.statistic.eventLog.events.EventFields
+import com.intellij.internal.statistic.eventLog.events.StringEventField
+import com.intellij.internal.statistic.service.fus.collectors.CounterUsagesCollector
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsMacroDefinitionBase
+
+@Suppress("UnstableApiUsage")
+class RsCodeVisionUsageCollector : CounterUsagesCollector() {
+    override fun getGroup(): EventLogGroup = GROUP
+
+    companion object {
+        fun logAuthorClicked(element: PsiElement) {
+            val location = getLocation(element)
+            AUTHOR_CLICKED_EVENT.log(element.project, location)
+        }
+
+        fun logUsagesClicked(element: PsiElement) {
+            val location = getLocation(element)
+            USAGES_CLICKED_EVENT_ID.log(element.project, location)
+        }
+
+        private fun getLocation(element: PsiElement): String {
+            return when (element) {
+                is RsFunction -> Location.FUNCTION
+                is RsStructItem -> Location.STRUCT_ITEM
+                is RsEnumItem -> Location.ENUM_ITEM
+                is RsEnumVariant -> Location.ENUM_VARIANT
+                is RsNamedFieldDecl -> Location.NAMED_FIELD_DECL
+                is RsTraitItem -> Location.TRAIT_ITEM
+                is RsImplItem -> Location.IMPL_ITEM
+                is RsTypeAlias -> Location.TYPE_ALIAS
+                is RsConstant -> Location.CONSTANT
+                is RsMacroDefinitionBase -> Location.MACRO_DEF
+                is RsModItem -> Location.MOD_ITEM
+                else -> Location.UNKNOWN
+            }.name
+        }
+
+        private val GROUP = EventLogGroup("rust.code.vision", 1)
+
+        private enum class Location(name: String) {
+            FUNCTION("function"),
+            STRUCT_ITEM("struct"),
+            ENUM_ITEM("enum"),
+            ENUM_VARIANT("enum_variant"),
+            NAMED_FIELD_DECL("field"),
+            TRAIT_ITEM("trait"),
+            IMPL_ITEM("impl"),
+            TYPE_ALIAS("type_alias"),
+            CONSTANT("constant"),
+            MACRO_DEF("macro_def"),
+            MOD_ITEM("mod_item"),
+            UNKNOWN("unknown"),
+        }
+
+        private val LOCATION_FIELD: StringEventField = EventFields.String(
+            "location",
+            Location.values().map { it.name }.toList()
+        )
```

Comment:
There are `EnumEventField` and `EventFields.Enum` not to convert enums to strings manually